service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java

```diff
@@ -281,12 +282,11 @@ public void specification(HttpServiceRequest request, final HttpServiceResponder
 
       Map<String, String> properties = new HashMap<>();
       properties.put("topic", topic);
-      properties.put("referenceName", topic);
+      properties.put("referenceName", topic + "_" + UUID.randomUUID());
```

Comment:
On second thoughts, it might actually be better to keep the reference name as topic only, and offset name unique, because the reference name shows up in lineage diagrams, etc. I can make that change.